### PR TITLE
add go template

### PR DESCRIPTION
**Pull request by Neeraj-Nec**
add gotemplate

### CODE REVIEW

**Review by showersky — APPROVED** [review]

File: application/template/NGSILD/goTemplate/function.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import (
+	"fmt"
+	//. "nec-fogflow/common/ngsi"
+)
+
+type publishContextFunc func(ctxObj map[string]interface{})
+
+func fogfunction(ctxObj map[string]interface{}, publish publishContextFunc) error {
+	fmt.Println(ctxObj)
+	//==============Implemet losic ==============
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/function.go:12]: [minor] it should be implementation logic

File: application/template/NGSILD/goTemplate/function.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package main
+
+import (
+	"fmt"
+	//. "nec-fogflow/common/ngsi"
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/function.go:5]: [minor] remove the commented code.

File: application/template/NGSILD/goTemplate/main.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+)
+
+//var ctxUpdateBuffer []*ContextObject
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/main.go:18]: [minor] remove the commented code.

File: application/template/NGSILD/goTemplate/main.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+	return commands
+}
+
+func startApp() {
+	fmt.Println("start to receive input data streams via a listening port")
+}
+
+func stopApp() {
+	fmt.Println("clean up the app")
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/main.go:49]: [minor] it should be stop and clean up the application

File: application/template/NGSILD/goTemplate/main.go [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+	handler for receiving notification
+*/
+func onNotify(w http.ResponseWriter, r *http.Request) {
+	if ctype := r.Header.Get("Content-Type"); ctype == "application/json" || ctype == "application/ld+json" {
+		//var context []interface{}
+		//context = append(context, DEFAULT_CONTEXT)
+		notifyElement, _ := getStringInterfaceMap(r)
+		notifyElemtData := notifyElement["data"]
+		notifyEleDatamap := notifyElemtData.([]interface{})
+		w.WriteHeader(201)
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/main.go:72]: [query] validate response should be 201 or 200

File: application/template/NGSILD/goTemplate/main.go [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+	for key, _ := range element {
+		if key != "id" && key != "type" && key != "modifiedAt" && key != "createdAt" && key != "observationSpace" && key != "operationSpace" && key != "location" && key != "@context" {
+			ctxObject[key] = element[key]
+		}
+	}
+	return ctxObject
+}
+
+func query2execution() map[string]interface{} {
+	//client := NGSILdClient{IoTBrokerURL: brokerURL}
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/main.go:166]: [minor] removed the commented codes

File: application/template/NGSILD/goTemplate/main.go [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+	ctxObjects, err := queryContext(inputEntityId, brokerURL)
+	if err != nil {
+		fmt.Println("failed to update context", err)
+		return nil
+	}
+	ctxObj := element2Object(ctxObjects)
+	return ctxObj
+}
+
+/*func handleCmds(commands []ConfigCommand) {
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/main.go:177]: [minor] remove the commented code

File: application/template/NGSILD/goTemplate/main.go [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+}
+
+func setReferenceURL(cmd map[string]interface{}) {
+	//myReferenceURL = cmd.ReferenceURL
+	//fmt.Println("your application can subscribe addtional inputs under the reference URL: " + myReferenceURL)
+	fmt.Println("Set ReferenceURL  in config.json")
+}
+
+//
+// publish context entities:
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/main.go:229]: [minor] remove unused code.

File: application/template/NGSILD/goTemplate/ngsiclient.go [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		return nil, err
+	}
+	itemsMap := reqData.(map[string]interface{})
+	fmt.Println(itemsMap)
+	res.Body.Close()
+	return itemsMap, err
+}
+
+func UpdateLdContext(updateCtx map[string]interface{}, IoTBrokerURL string) error {
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/ngsiclient.go:35]: [minor] Function specification is missing. Need to explain what this function is used for.

File: application/template/NGSILD/goTemplate/config.json [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+{
+	"command":"CONNECT_BROKER",
+	"brokerURL": "http://180.179.214.202:8070" 
+},
+{
+	"command":"SET_INPUTS",
+	"type": "test",
+	"id":   "urn:ngsi-ld:Vehicle:A14"
+}
+]
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/config.json:11]: [Query] Do SET_OUTPUT not needed here

File: application/template/NGSILD/goTemplate/ngsiclient.go [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+func queryContext(id string, IoTBrokerURL string) (map[string]interface{}, error) {
```
Comment (naveennec) [inline application/template/NGSILD/goTemplate/ngsiclient.go:11]: [minor] Function specification is missing here. Mention what is the use of this function.

File: application/template/NGSILD/goTemplate/function.go (hunk 1 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/function.go:12]: Done

File: application/template/NGSILD/goTemplate/main.go (hunk 3 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/main.go:18]: removed

File: application/template/NGSILD/goTemplate/main.go (hunk 4 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/main.go:49]: Done

File: application/template/NGSILD/goTemplate/ngsiclient.go (hunk 9 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/ngsiclient.go:35]: done

File: application/template/NGSILD/goTemplate/main.go (hunk 8 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/main.go:229]: done

File: application/template/NGSILD/goTemplate/config.json (hunk 10 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/config.json:11]: Done

File: application/template/NGSILD/goTemplate/ngsiclient.go (hunk 11 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/ngsiclient.go:11]: Done

File: application/template/NGSILD/goTemplate/main.go (hunk 6 above)
Comment (Neeraj-Nec) [reply application/template/NGSILD/goTemplate/main.go:166]: Done

**Pull request merged**